packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -78,11 +80,20 @@ export async function createIncomingPaymentService(
 async function getIncomingPayment(
   deps: ServiceDependencies,
   key: string,
-  value: string
+  value: string,
+  clientId?: string
 ): Promise<IncomingPayment | undefined> {
-  const incomingPayment = await IncomingPayment.query(deps.knex)
-    .findOne(key, value)
-    .withGraphFetched('asset')
+  let incomingPayment: IncomingPayment
+  if (!clientId) {
+    incomingPayment = await IncomingPayment.query(deps.knex)
+      .findOne(key, value)
+      .withGraphFetched('asset')
+  } else {
+    incomingPayment = await IncomingPayment.query(deps.knex)
+      .findOne(`incomingPayments.${key}`, value)
+      .withGraphJoined('[asset, grant]')
+      .where('grant.clientId', clientId)
```

Comment:
I thought about that, too, and thought that the routes don't return it because it's not included in the `toJSON` and it's also not included in the` OutgoingPayment` type on the admin API.